pkg/cmd/cli/cmd/env.go

```diff
@@ -268,14 +285,14 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 		}
 	}
 	if one && skipped == len(infos) {
-		return fmt.Errorf("the %s %s is not a pod or does not have a pod template", infos[0].Mapping.Resource, infos[0].Name)
+		return fmt.Errorf("%s/%s is not a pod or does not have a pod template", infos[0].Mapping.Resource, infos[0].Name)
 	}
 
 	if list {
 		return nil
 	}
 
-	objects, err := resource.AsVersionedObject(infos, false, outputVersion)
+	objects, err := resource.AsVersionedObjects(infos, outputVersion)
```

Comment:
That's good. I am thinking that --api-version informs the negotiation that results in clientConfig.Version… we may just be able to use that. Not sure how to really test that if we only support one API version currently